nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/QueryRecord.java

```diff
@@ -640,4 +667,233 @@ public Connection getConnection() {
             return connection;
         }
     }
+
+
+    // ------------------------------------------------------------
+    // User-Defined Functions for Calcite
+    // ------------------------------------------------------------
+
+
+    public static class ObjectRecordPath extends RecordPathFunction {
```

Comment:
These static classes might be better off in a util class (like we do with `StandardValidators`) than in a single processor

Reply:
They could be moved out if ever we found a reason to reference them elsewhere. However, I do generally agree with the concept outlined in Joshua Bloch's "Effective Java", that classes, in general, should be made inner classes if they will not have any applicability outside of the outer class. In this case, I don't think it would make sense to use the class outside of QueryRecord.